File: 50dias/python/45.py
import random

def EmbaralhaPalavraMantendoPrimeiraUltima(palavra):
    tamanhoPalavra = len(palavra)
    novaPalavra = palavra
    if tamanhoPalavra > 1:
        meio = palavra[1:tamanhoPalavra-1]
        novoMeio = ''.join(random.sample(meio, len(meio)))
        novaPalavra = palavra[0] + novoMeio + palavra[tamanhoPalavra-1]
    return novaPalavra

palavra = input('Digite uma palavra: ')
palavra = EmbaralhaPalavraMantendoPrimeiraUltima(palavra)
print(palavra)

File: 50dias/cpp/45.cpp
#include <iostream>
#include <random>

using namespace std;

void EmbaralhaPalavraMantendoPrimeiraUltimaLetra(
    string& palavra
)
{
    std::random_device rd;
    std::mt19937 g(rd());

    std::shuffle(++palavra.begin(),
        --palavra.end(),
        g);
}

int main()
{
    string palavra;
    cout << "Digite uma palavra: ";
    cin >> palavra;

    if (palavra.size() > 1) {
        EmbaralhaPalavraMantendoPrimeiraUltimaLetra(palavra);
    }

    cout << endl << palavra << endl;
    return 0;
}